lib/style-transform.js

```diff
@@ -1,407 +1,77 @@
-                                            // ignore
-                                            continue
-                                          } else {
-                                            _line += piece + suffix + ' '
-                                            piece = ''
-                                          }
-                                        } else if (chr === ':') {
-                                          // pseudo-class or preudo-element
-                                          _line += piece + suffix
-                                          piece = chr
-                                          inCombinatorOrPseudo = true
-                                        } else if (chr === '+' || chr === '~' || chr === '>') {
-                                          // combinators
-                                          piece += chr
-                                          inCombinatorOrPseudo = true
-                                        } else {
-                                          piece += chr
-                                        }
-                                      }
-                                    }
-                                  }
-                                }
-
-                                // flush remainder
-                                if (piece.length) {
-                                  _line += piece
-                                  if (!inGlobal && !inCombinatorOrPseudo) {
-                                    _line += suffix
-                                  }
-                                }
-
-                                if (isLast) {
-                                  _line += '{'
-                                } else {
-                                  _line += ','
-                                }
-                            }
-
-                            line = _line;
-                        }
-                    }
-                }
-
-                generator && generator.addMapping({
-                    generated: {
-                      line: 1,
-                      column: output.length
-                    },
-                    source: fileName,
-                    original: {
-                        line: lineNumber,
-                        column: columnNumber
-                    }
-                })
-
-                output += line;
-                line = '';
-            }
-            // not `\t`, `\r`, `\n` characters
-            else if (code !== 9 && code !== 13 && code !== 10) {
-                line += styles[i];
-            }
-
-            // next character
-            i++;
-            columnNumber++;
-        }
-
-        return output;
+function replace (match) {
+  // :hover etc...
+  if ((indexOf = match.indexOf(':')) !== -1) {
+    // tries to matche `g` character in :global()
+    if (match.charCodeAt(indexOf + 1) === 103) {
+      return match.replace(regExpGlobal, '$1').trim()
```

Comment:
I have mixed feelings about regexp globals, maybe you can use a function as 2nd argument for `replace`?